### PR TITLE
Implement ALL view for updates; Integrate projects/invites with backend; Fix misc bugs

### PR DESCRIPTION
**Pull request by amy-lei**
- Implemented`All` view for updates
- Fixed bugs with project backend
- Fixed bug with user collection that queried by username instead of email
- Made POST route for projects take in a list of emails instead and update validators
- Replaced projects/invites frontend with backend

image

### CODE REVIEW

**Comment by opheeelia** [comment]
Wait I thought for ALL view it wouldn't group by users? I was imagining just getting rid of the user headers and having each update say its author?

**Comment by amy-lei** [comment]
> Wait I thought for ALL view it wouldn't group by users? I was imagining just getting rid of the user headers and having each update say its author?

With this approach, I was thinking we'd limit to most recent X updates and then have a "see more..." button that when clicked, directs you to that user's view.

But that works too! I don't quite remember what everyone agreed upon in the end was

File: client/components/Project/AddProjectComponent.vue [hunk 1]
```diff
@@ -103,9 +103,35 @@ export default {
...
+        body: JSON.stringify({
+          projectName: this.fields.name,
+          scheduledUpdates: [this.fields.deadline],
+          invitedUsers: this.fields.collaborators,
+        }),
+      };
+      try {
+        const res = await fetch('/api/projects', options);
+        if (!res.ok) {
+          throw Error(resJson.error);
```
Comment (opheeelia) [inline client/components/Project/AddProjectComponent.vue:120]: resJson is not defined here!

File: client/components/Project/AddProjectComponent.vue [hunk 2]
```diff
@@ -103,9 +103,35 @@ export default {
...
-      console.log(this.fields);
-      this.hideModal();
+    async submit() {
+      const options = {
+        method: 'POST',
+        headers: {'Content-Type': 'application/json'},
+        credentials: 'same-origin',
+        body: JSON.stringify({
+          projectName: this.fields.name,
+          scheduledUpdates: [this.fields.deadline],
```
Comment (maoivy) [inline client/components/Project/AddProjectComponent.vue:113]: Scheduled updates are optional right? I'm getting a 400 bad request when I try to make a project without a deadline, and it seems like in the frontend it's not expecting the possibility of it either (says reading length of undefined?)

**Review by opheeelia — CHANGES_REQUESTED** [review]

File: client/components/Project/AddProjectComponent.vue [hunk 3]
```diff
@@ -103,9 +103,35 @@ export default {
...
+          invitedUsers: this.fields.collaborators,
```
Comment (maoivy) [inline client/components/Project/AddProjectComponent.vue:114]: Similar to the deadline one, this.fields.collaborators will have one empty string in it if collaborators is empty (and I think we should allow for no collaborators)

File: client/components/Project/AddProjectComponent.vue (hunk 2 above)
Comment (amy-lei) [reply client/components/Project/AddProjectComponent.vue:113]: Resolved offline!

File: client/components/Project/AddProjectComponent.vue (hunk 3 above)
Comment (amy-lei) [reply client/components/Project/AddProjectComponent.vue:114]: The user in this case would just delete the entry before submitting. We will eventually have validators on the frontend too, preventing the submission of empty strings.

File: client/components/Project/AddProjectComponent.vue (hunk 1 above)
Comment (amy-lei) [reply client/components/Project/AddProjectComponent.vue:121]: Good catch!

**Review by opheeelia — APPROVED** [review]

**Pull request merged**